internal/pkg/mount/v3/helpers.go

```diff
@@ -0,0 +1,313 @@
+}
+
+// Squashfs binds the squashfs to the loop device and returns the mountpoint for it to the specified target.
+func Squashfs(target, squashfsFile string, printer func(string, ...any)) (*Manager, error) {
+	dev, err := losetup.Attach(squashfsFile, 0, true)
+	if err != nil {
+		return nil, err
+	}
+
+	return NewManager(
+		WithTarget(target),
+		WithPrinter(printer),
+		WithReadOnly(),
+		WithShared(),
+		WithExtraUnmountCallbacks(func(m *Manager) {
+			dev.Detach() //nolint:errcheck
+		}),
+		WithFsopen(
+			"squashfs",
+			fsopen.WithSource(dev.Path()),
+			fsopen.WithBoolParameter("ro"),
+			fsopen.WithPrinter(printer),
+		),
+	), nil
+}
+
+func gather[T comparable](c ...func() T) []T {
+	var (
+		zero T
+		vals []T
+	)
+
+	for _, f := range c {
+		val := f()
+
+		if val != zero {
+			vals = append(vals, val)
+		}
+	}
+
+	return vals
+}
+
+func newManager(condition func() bool, opts ...ManagerOption) func() *Manager {
+	return func() *Manager {
+		if !condition() {
+			return nil
+		}
+
+		return NewManager(opts...)
+	}
+}
+
+func always() bool { return true }
+
+func hasEFIVars() bool {
+	_, err := os.Stat(constants.EFIVarsMountPoint)
+
+	if err == nil || errors.Is(err, os.ErrNotExist) {
+		return err == nil
+	}
+
+	// this means something else is wrong, let's panic
+	// as this shoul never happen
+	panic(err)
+}
+
+// Pseudo creates basic filesystem mountpoint managers.
+func Pseudo(printer func(string, ...any)) Managers {
+	return gather(
+		newManager(
+			always,
+			WithTarget("/dev"),
+			WithKeepOpenAfterMount(),
+			WithMountAttributes(unix.MOUNT_ATTR_NOSUID),
+			WithFsopen(
+				"devtmpfs",
+				fsopen.WithStringParameter("mode", "0755"),
+				fsopen.WithPrinter(printer),
+			),
+		),
+		newManager(
+			always,
+			WithTarget("/proc"),
+			WithKeepOpenAfterMount(),
+			WithMountAttributes(unix.MOUNT_ATTR_NOSUID|unix.MOUNT_ATTR_NOEXEC|unix.MOUNT_ATTR_NODEV),
+			WithFsopen("proc", fsopen.WithPrinter(printer)),
+		),
+		newManager(
+			always,
+			WithTarget("/sys"),
+			WithKeepOpenAfterMount(),
+			WithFsopen("sysfs", fsopen.WithPrinter(printer)),
+		),
+	)
+}
+
+// PseudoLate creates late pseudo filesystem mountpoint managers.
+func PseudoLate(printer func(string, ...any)) Managers {
+	return gather(
+		newManager(
+			always,
+			WithTarget("/run"),
+			WithMountAttributes(unix.MOUNT_ATTR_NOSUID|unix.MOUNT_ATTR_NOEXEC|unix.MOUNT_ATTR_RELATIME),
+			WithSelinuxLabel(constants.RunSelinuxLabel),
+			WithFsopen(
+				"tmpfs",
+				fsopen.WithPrinter(printer),
+				fsopen.WithStringParameter("mode", "0755"),
+			),
+		),
+		newManager(
+			always,
+			WithTarget("/system"),
+			WithSelinuxLabel(constants.SystemSelinuxLabel),
+			WithFsopen(
+				"tmpfs",
+				fsopen.WithPrinter(printer),
+				fsopen.WithStringParameter("mode", "0755"),
+			),
+		),
+		newManager(
+			always,
+			WithTarget("/tmp"),
+			WithMountAttributes(unix.MOUNT_ATTR_NOSUID|unix.MOUNT_ATTR_NOEXEC|unix.MOUNT_ATTR_NODEV),
+			WithFsopen(
+				"tmpfs",
```

Comment:
https://www.kernel.org/doc/html/latest/filesystems/tmpfs.html I wonder if we should also do tmpfs quotas, something for later not this PR, FYI @smira

Reply:
yeah, quotas might be cool, not sure if we have any way to enforce them? (most should come from `root` anyways?)